ibverbs/src/lib.rs

```diff
@@ -1488,26 +1530,16 @@ impl<'res> QueuePair<'res> {
     ///
     /// [1]: http://www.rdmamojo.com/2013/01/26/ibv_post_send/
     #[inline]
-    pub unsafe fn post_send<T, R>(
+    pub unsafe fn post_send<'a>(
         &mut self,
-        mr: &mut MemoryRegion<T>,
-        range: R,
+        local: &[LocalMemorySlice<'a>],
         wr_id: u64,
-    ) -> io::Result<()>
-    where
-        R: sliceindex::SliceIndex<[T], Output = [T]>,
```

Comment:
The reason why `SliceIndex` is nice is because it also works for things like solitary `usize`. It's a little sad to lose that for ergonomics, though I suppose it's not the end of the world if it enables a class of use-cases that were otherwise inaccessible.